gremlin-go/driver/client.go

```diff
@@ -84,12 +95,21 @@ func (client *Client) Submit(traversalString string) (ResultSet, error) {
 	// TODO AN-982: Obtain connection from pool of connections held by the client.
 	client.logHandler.logf(Debug, submitStartedString, traversalString)
 	request := makeStringRequest(traversalString, client.traversalSource)
+	// TODO: Add bindings to request. request.args['bindings'] = bindings
+	addSessionArgsToRequest(client, request)
```

Comment:
1. This should not take in the client. The `if` check needs to still be here.
2. You're passing the request by value in `addSessionArgsToRequest`, so modifying it won't work